ksqldb-engine/src/test/java/io/confluent/ksql/topic/TopicCreateInjectorTest.java

```diff
@@ -532,6 +584,74 @@ public void shouldThrowIfRetentionConfigPresentInCreateTableAs() {
             + " Non-windowed tables do not support retention."));
   }
 
+  @Test
+  public void shouldThrowIfCleanupPolicyPresentInCreateTable() {
+    // Given:
+    givenStatement("CREATE TABLE foo_bar (FOO STRING PRIMARY KEY, BAR STRING) WITH (kafka_topic='foo', partitions=1, cleanup_policy='whatever');");
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> injector.inject(statement, builder)
+    );
+
+    // Then:
+    assertThat(
+        e.getMessage(),
+        containsString("Invalid config variable in the WITH clause: CLEANUP_POLICY"));
+  }
+
+  @Test
+  public void shouldThrowIfCleanupPolicyConfigPresentInCreateTableAs() {
+    // Given:
+    givenStatement("CREATE TABLE foo_bar WITH (kafka_topic='foo', partitions=1, cleanup_policy='whatever') AS SELECT * FROM SOURCE;");
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> injector.inject(statement, builder)
+    );
+
+    // Then:
+    assertThat(
+        e.getMessage(),
+        containsString("Invalid config variable in the WITH clause: CLEANUP_POLICY"));
+  }
+
+  @Test
+  public void shouldThrowIfCleanupPolicyConfigPresentInCreateStream() {
+    // Given:
+    givenStatement("CREATE STREAM x (FOO VARCHAR) WITH (kafka_topic='foo', partitions=1, cleanup_policy='whatever');");
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> injector.inject(statement, builder)
+    );
+
+    // Then:
+    assertThat(
+        e.getMessage(),
+        containsString("Invalid config variable in the WITH clause: CLEANUP_POLICY"));
+  }
+
+  @Test
+  public void shouldThrowIfCleanupPolicyConfigPresentInCreateStreamAs() {
+    // Given:
+    givenStatement("CREATE STREAM x WITH (kafka_topic='topic', partitions=1, cleanup_policy='whatever') AS SELECT * FROM SOURCE;");
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> injector.inject(statement, builder)
+    );
+
+    // Then:
+    assertThat(
+        e.getMessage(),
+        containsString("Invalid config variable in the WITH clause: CLEANUP_POLICY"));
```

Comment:
we should be blocking users from setting it manually as well, right?

Reply:
yes, I think it would be error-prone to allow people to change this configuration